Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-234.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 1641f7d5cae413d7021a04609e7ae1b4bbfb0430
Containerized:False
Number of Error(s): 52
/home/centos/smoketest/PR-15748/HPCC-Platform/common/remote/hooks/git/libgit2/src/common.h:14:10: fatal error: git2/common.h: No such file or directory
make[2]: *** [common/remote/hooks/git/libgit2/src/CMakeFiles/git2internal.dir/build.make:1336: common/remote/hooks/git/libgit2/src/CMakeFiles/git2internal.dir/path.c.o] Error 1
cc: error: /home/centos/smoketest/PR-15748/HPCC-Platform/common/remote/hooks/git/libgit2/src/pathspec.c: No such file or directory
cc: fatal error: no input files
make[2]: *** [common/remote/hooks/git/libgit2/src/CMakeFiles/git2internal.dir/build.make:1350: common/remote/hooks/git/libgit2/src/CMakeFiles/git2internal.dir/pathspec.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:9318: common/remote/hooks/git/libgit2/src/CMakeFiles/git2internal.dir/all] Error 2
cannot fetch initial working directory: No such file or directory at /usr/share/autoconf/Autom4te/General.pm line 196.
make[2]: *** [plugins/cassandra/CMakeFiles/libuv.dir/build.make:73: plugins/cassandra/.libs/libuv.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:15575: plugins/cassandra/CMakeFiles/libuv.dir/all] Error 2
CMake Error: File /home/centos/smoketest/PR-15748/HPCC-Platform/plugins/couchbase/libcouchbase/cmake/config-cmake.h.in does not exist.
CMake Error at cmake/Modules/GenerateConfigDotH.cmake:47 (CONFIGURE_FILE):
checking for libpthread (by pkg-config)...-- Configuring incomplete, errors occurred!
make[2]: *** [plugins/couchbase/CMakeFiles/libcouchbase-build.dir/build.make:74: plugins/couchbase/libcouchbase-build/lib/libcouchbase.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:15891: plugins/couchbase/CMakeFiles/libcouchbase-build.dir/all] Error 2
CMake Error: The source directory /home/centos/smoketest/PR-15748/HPCC-Platform/system/logaccess/ElasticStack/elasticlient does not appear to contain CMakeLists.txt.
gmake[4]: *** [CMakeFiles/Makefile2:309: cmake_check_build_system] Error 1
gmake[3]: *** [Makefile:169: jsoncpp_lib] Error 2
make[2]: *** [system/logaccess/ElasticStack/CMakeFiles/elasticlient-build.dir/build.make:86: system/logaccess/ElasticStack/elasticlient-build-prefix/src/elasticlient-build-stamp/elasticlient-build-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:7394: system/logaccess/ElasticStack/CMakeFiles/elasticlient-build.dir/all] Error 2
make[2]: *** [CMakeFiles/generate-boost-regex.dir/build.make:92: generate-boost-regex-prefix/src/generate-boost-regex-stamp/generate-boost-regex-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:7097: CMakeFiles/generate-boost-regex.dir/all] Error 2
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
chdir: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
/bin/sh: line 0: cd: /home/centos/smoketest/PR-15748/build/system/aws/aws-sdk-cpp/.deps/build/src/AwsCCommon-build/tests: No such file or directory
gmake[8]: *** [tests/CMakeFiles/aws-c-common-tests.dir/build.make:230: tests/CMakeFiles/aws-c-common-tests.dir/date_time_test.c.o] Error 1
gmake[7]: *** [CMakeFiles/Makefile2:910: tests/CMakeFiles/aws-c-common-tests.dir/all] Error 2
gmake[6]: *** [Makefile:166: all] Error 2
gmake[5]: *** [CMakeFiles/AwsCCommon.dir/build.make:86: build/src/AwsCCommon-stamp/AwsCCommon-build] Error 2
gmake[4]: *** [CMakeFiles/Makefile2:87: CMakeFiles/AwsCCommon.dir/all] Error 2
gmake[3]: *** 
 ... ( comment is too long, 384 lines, 31813 bytes, truncated) ... 
[Build error:](https://gist.github.com/dafd0debb26b984599554127af0640f9)
Build log [Build log](https://gist.github.com/HPCCSmoketest/dafd0debb26b984599554127af0640f9#file-result-22-02-09-13-27-23-log)

```

--- a/system/security/cryptohelper/digisign.cpp
+++ b/system/security/cryptohelper/digisign.cpp
@@ -206,7 +206,7 @@ public:
 
 
 static IDigitalSignatureManager * dsm = nullptr;
-#if defined(_USE_OPENSSL) && !defined(_WIN32)
+#if defined(_USE_OPENSSL)
 static std::once_flag dsmInitFlag;
 #endif
 
@@ -219,7 +219,7 @@ MODULE_EXIT()
     ::Release(dsm);
 }
 
-#if defined(_USE_OPENSSL) && !defined(_WIN32)
+#if defined(_USE_OPENSSL)
 static void createDigitalSignatureManagerInstance(IDigitalSignatureManager * * ppDSM)
 {
     const char * pubKey = nullptr, *privKey = nullptr, *passPhrase = nullptr;
@@ -238,7 +238,7 @@ static void createDigitalSignatureManagerInstance(IDigitalSignatureManager * * p
 //Returns reference to singleton instance created from environment.conf key file settings
 IDigitalSignatureManager * queryDigitalSignatureManagerInstanceFromEnv()
 {
-#if defined(_USE_OPENSSL) && !defined(_WIN32)
+#if defined(_USE_OPENSSL)
     std::call_once(dsmInitFlag, createDigitalSignatureManagerInstance, &dsm);
     return dsm;
 #else
@@ -255,7 +255,7 @@ IDigitalSignatureManager * createDigitalSignatureManagerInstanceFromFiles(const 
 
 IDigitalSignatureManager * createDigitalSignatureManagerInstanceFromFiles(const char * pubKeyFileName, const char *privKeyFileName, size32_t lenPassphrase, const void * passPhrase)
 {
-#if defined(_USE_OPENSSL) && !defined(_WIN32)
+#if defined(_USE_OPENSSL)
     Owned<CLoadedKey> pubKey, privKey;
     Owned<IMultiException> exceptions;
     if (!isEmptyString(pubKeyFileName))
@@ -314,7 +314,7 @@ IDigitalSignatureManager * createDigitalSignatureManagerInstanceFromKeys(const c
 
 IDigitalSignatureManager * createDigitalSignatureManagerInstanceFromKeys(const char * pubKeyString, const char * privKeyString, size32_t lenPassphrase, const void * passPhrase)
 {
-#if defined(_USE_OPENSSL) && !defined(_WIN32)
+#if defined(_USE_OPENSSL)
     Owned<CLoadedKey> pubKey, privKey;
 
     Owned<IMultiException> exceptions;

